bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/i18n/DiscoveryResultI18nUtil.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.config.discovery.i18n;
+
+import java.util.Locale;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.internal.DiscoveryResultImpl;
+import org.eclipse.smarthome.core.i18n.I18nUtil;
+import org.eclipse.smarthome.core.i18n.TranslationProvider;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+
+/**
+ * The {@link DiscoveryResultI18nUtil} can be used to localize the {@link DiscoveryResult} of a thing discovery
+ * using the I18N mechanism of the Eclipse SmartHome framework. Currently the label of the {@link DiscoveryResult}
+ * is the single attribute which can be localized.
+ * It automatically infers the key if the default text is not a constant.
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+public class DiscoveryResultI18nUtil {
+
+    private TranslationProvider i18nProvider;
+
+    public DiscoveryResultI18nUtil(TranslationProvider i18nProvider) {
+        this.i18nProvider = i18nProvider;
+    }
+
+    public DiscoveryResult createLocalizedDiscoveryResult(AbstractDiscoveryService discoveryService,
+            DiscoveryResult discoveryResult, Locale locale) {
+        Bundle bundle = FrameworkUtil.getBundle(discoveryService.getClass());
+
+        String defaultLabel = discoveryResult.getLabel();
+
+        String key = I18nUtil.isConstant(defaultLabel) ? I18nUtil.stripConstant(defaultLabel)
+                : inferKey(discoveryResult, "label");
+
+        String label = i18nProvider.getText(bundle, key, defaultLabel, locale);
+
+        return new DiscoveryResultImpl(discoveryResult.getThingTypeUID(), discoveryResult.getThingUID(),
+                discoveryResult.getBridgeUID(), discoveryResult.getProperties(),
+                discoveryResult.getRepresentationProperty(), label, discoveryResult.getTimeToLive());
+    }
+
+    private String inferKey(DiscoveryResult discoveryResult, String lastSegment) {
+        return "discovery." + discoveryResult.getThingUID().getAsString().replaceAll(":", ".") + "." + lastSegment;
```

Comment:
Using the ThingUID works for the bindings you selected. For the WeMo binding this will not work, because the ThingUID will be different for each thing, but the label would need translation independently of that ([see here](https://github.com/eclipse/smarthome/blob/7fe6f9081a0c5800ee8bbe968d22f7fab5db96a3/extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/discovery/WemoDiscoveryParticipant.java#L44)).

Reply:
For WeMo like for Sonos for example, the used label is built from the product name returned by UPnP. I don't see a real need for translation. How could you anticipate all possible discovered things ?
I am clearly not covering this.

Reply:
WeMo uses a default label "WeMo device" if nothing specific is obtained from the device. This could be translated. I'm working on a OneWire binding, discovered temnperature sensors get the label "temperature sensor" or something like that, this oculd be translated, too.

Reply:
Ok with you. It should be possible to translate the default label. I think it could be done without any change to my PR by using @text/xxx as label. I will check that.

Reply:
... by using 
```
@text/xxx
```

Reply:
Ok, I can confirm that it is possible to use
```
"@text/discovery.ntp.defaultLabel"
```
as label in the binding and define for example in the property file:
```
discovery.ntp.defaultLabel = here is a test
```